### PR TITLE
Rework Flock compute updates

### PR DESCRIPTION
**Pull request by FlameArrow57**
[REWORK][QOL][PERFORMANCE][BUG]

## About the PR 
This PR reworks how Flock used and total compute is updated for Flockmind and Flocktraces.

Compute will now only be updated when a thing that is added or removed from a Flock actually modifies compute.

Compute will also now only be updated corresponding to the change added by the thing being added or removed from the Flock instead of compute being recalculated iterating over every Flock thing.

Compute for Flocktraces will now also update along with the Flockmind's compute updates.

Fixes compute not updating when an amount is used or given by Flock structures.

## Why's this needed? 
Current compute updates update compute looking at everything in the Flock instead of just what is being added or removed.

Flocktrace compute updating was done in the life loop, which would mean updates weren't as often.

Fixes related bug.

### CODE REVIEW

**Review by amylizzle — CHANGES_REQUESTED** [review]
One major but fixable problem with this: Some structures (ie, Sentinels) vary the amount of compute they use depending on behaviour. You'd need to get them to send an update when compute usage changed. It probably wouldn't be a terrible idea to make `changeCompute()` a proc on flock_structure parent, and handle the update there.

File: code/datums/flock/flock.dm [hunk 1]
```diff
@@ -186,44 +188,24 @@ var/flock_signal_unleashed = FALSE
 
 
 /datum/flock/proc/total_compute()
```
[the remark this answers is not in the document]
Comment (FlameArrow57) [reply code/datums/flock/flock.dm:190]: There's several places where it is though, so was uncertain

**Comment by TobleroneSwordfish** [comment]
Does this fix #9641 or is that different spaget?

File: code/datums/flock/flock.dm [hunk 2]
```diff
@@ -186,44 +188,24 @@ var/flock_signal_unleashed = FALSE
...
-			if(comp_provided<0)
-				. += abs(comp_provided)
+/datum/flock/proc/can_afford_compute(var/cost)
+	return (cost <= src.total_compute() - src.used_compute)
 
-	for(var/obj/flock_structure/S as anything in src.structures)
-		comp_provided = S.compute_provided()
-		if(comp_provided<0)
-			. += abs(comp_provided)
+/datum/flock/proc/update_computes()
```
[the remark this answers is not in the document]
Comment (FlameArrow57) [reply code/datums/flock/flock.dm:199]: Idk, they are on different types, on datum versus ability holder

**Comment by FlameArrow57** [comment]
It does not, wasn't able to fix that

**Comment by ZeWaka** [comment]
x

**Comment by FlameArrow57** [comment]
Currently a small bug with this, will try to fix in the next 24 hours

**Comment by FlameArrow57** [comment]
Ok should be ready

**Pull request merged**